src/common/utils/reactUtils 2.ts

```diff
@@ -0,0 +1,7 @@
+import { useMemo } from "react";
+import { useLocation } from "react-router";
+
+export function useQuery() {
+  const { search } = useLocation();
+  return useMemo(() => new URLSearchParams(search), [search]);
+}
```

Comment:
🔧 We can use the react-router-dom provided function `useSearchParams` to to get this result